recipes/soci/all/test_package/CMakeLists.txt

```diff
@@ -1,62 +1,35 @@
 cmake_minimum_required(VERSION 3.0)
 project(SociTest)
-enable_testing()
 
-set(CMAKE_BUILD_TYPE "Debug")
-
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-
-conan_basic_setup(TARGETS)
-conan_include_build_modules()
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup()
 
 if (WITH_EMPTY)
-    add_executable(empty_exe soci.cpp)
-    target_compile_definitions(empty_exe PUBLIC BACKEND_EMPTY)
-    target_link_libraries(empty_exe PUBLIC SOCI::soci_core SOCI::soci_empty)
-    add_test(
-        NAME TestSociEmpty
-        COMMAND empty_exe
-    )
-
+    add_executable(test_empty soci.cpp)
+    target_compile_definitions(test_empty PRIVATE BACKEND_EMPTY)
+    target_link_libraries(test_empty PUBLIC "${CONAN_LIBS}") #FIXME: use components SOCI::soci_empty)
```

Comment:
@madebr The issue here is that component dependencies are not well ordered :
`root@81d62ae04170:/develop/src/recipes/soci/all/test_package/build/f1a8359cf87cdfff32f468ea8ae5430f761c4908# /usr/bin/c++    -m64 -O3 -DNDEBUG        -rdynamic CMakeFiles/test_postgresql.dir/postgresql.cpp.o  -o bin/test_postgresql  -L/root/.conan/data/soci/4.0.0/_/_/package/289a71084365267bf537aa107e835a64d248603b/lib  -L/root/.conan/data/sqlite3/3.31.1/_/_/package/0ea13fbeff7afbe2f19b5c349acf59a585a00e93/lib  -L/root/.conan/data/libmysqlclient/8.0.17/_/_/package/bfe11e103319523e77cc827e33f3e1d8d2fa3a63/lib  -L/root/.conan/data/odbc/2.3.7/_/_/package/b29d3eb003873b92a248c0df6debab47f53853ea/lib  -L/root/.conan/data/libpq/12.2/_/_/package/8e0939db49a1d312829524beb4d0b6824e47691d/lib  -L/root/.conan/data/openssl/1.1.1g/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib  -L/root/.conan/data/zlib/1.2.11/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib  -L/root/.conan/data/libiconv/1.16/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib -Wl,-rpath,/root/.conan/data/soci/4.0.0/_/_/package/289a71084365267bf537aa107e835a64d248603b/lib:/root/.conan/data/sqlite3/3.31.1/_/_/package/0ea13fbeff7afbe2f19b5c349acf59a585a00e93/lib:/root/.conan/data/libmysqlclient/8.0.17/_/_/package/bfe11e103319523e77cc827e33f3e1d8d2fa3a63/lib:/root/.conan/data/odbc/2.3.7/_/_/package/b29d3eb003873b92a248c0df6debab47f53853ea/lib:/root/.conan/data/libpq/12.2/_/_/package/8e0939db49a1d312829524beb4d0b6824e47691d/lib:/root/.conan/data/openssl/1.1.1g/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib:/root/.conan/data/zlib/1.2.11/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib:/root/.conan/data/libiconv/1.16/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib -lsoci_core -lsoci_empty -lsoci_sqlite3 -lsoci_mysql -lsoci_odbc -lsoci_postgresql -lsqlite3 -lmysqlclient -lodbc -lodbccr -lodbcinst -lltdl -ldl -lpgcommon -lpgcommon_shlib -lpq -lssl -lcrypto -lz -liconv -lm -ldl -lpthread -lpgcommon -lpgcommon_shlib -lpq -lssl -lcrypto -lz -liconv -lm -lpthread`
When I would need this (-lsoci_core after all the backends)
`root@81d62ae04170:/develop/src/recipes/soci/all/test_package/build/f1a8359cf87cdfff32f468ea8ae5430f761c4908# /usr/bin/c++    -m64 -O3 -DNDEBUG        -rdynamic CMakeFiles/test_postgresql.dir/postgresql.cpp.o  -o bin/test_postgresql  -L/root/.conan/data/soci/4.0.0/_/_/package/289a71084365267bf537aa107e835a64d248603b/lib  -L/root/.conan/data/sqlite3/3.31.1/_/_/package/0ea13fbeff7afbe2f19b5c349acf59a585a00e93/lib  -L/root/.conan/data/libmysqlclient/8.0.17/_/_/package/bfe11e103319523e77cc827e33f3e1d8d2fa3a63/lib  -L/root/.conan/data/odbc/2.3.7/_/_/package/b29d3eb003873b92a248c0df6debab47f53853ea/lib  -L/root/.conan/data/libpq/12.2/_/_/package/8e0939db49a1d312829524beb4d0b6824e47691d/lib  -L/root/.conan/data/openssl/1.1.1g/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib  -L/root/.conan/data/zlib/1.2.11/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib  -L/root/.conan/data/libiconv/1.16/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib -Wl,-rpath,/root/.conan/data/soci/4.0.0/_/_/package/289a71084365267bf537aa107e835a64d248603b/lib:/root/.conan/data/sqlite3/3.31.1/_/_/package/0ea13fbeff7afbe2f19b5c349acf59a585a00e93/lib:/root/.conan/data/libmysqlclient/8.0.17/_/_/package/bfe11e103319523e77cc827e33f3e1d8d2fa3a63/lib:/root/.conan/data/odbc/2.3.7/_/_/package/b29d3eb003873b92a248c0df6debab47f53853ea/lib:/root/.conan/data/libpq/12.2/_/_/package/8e0939db49a1d312829524beb4d0b6824e47691d/lib:/root/.conan/data/openssl/1.1.1g/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib:/root/.conan/data/zlib/1.2.11/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib:/root/.conan/data/libiconv/1.16/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib -lsoci_empty -lsoci_sqlite3 -lsoci_mysql -lsoci_odbc -lsoci_postgresql -lsoci_core -lsqlite3 -lmysqlclient -lodbc -lodbccr -lodbcinst -lltdl -ldl -lpgcommon -lpgcommon_shlib -lpq -lssl -lcrypto -lz -liconv -lm -ldl -lpthread -lpgcommon -lpgcommon_shlib -lpq -lssl -lcrypto -lz -liconv -lm -lpthread`

Reply:
Oh, if that's correct, that is a bug in conan components and you should open a issue at https://github.com/conan-io/conan/issues

@danimtb 
Do conan components respect the dependency tree?

Reply:
@syoliver Can you please update your conan version to the latest (1.25.1) ?
conan components are a new feature and experimental so maybe you're experiencing a bug  that is already fixed.

Reply:
Yes @madebr it works fine with latest conan version, sorry for that.